hvplot/cli.py

```diff
@@ -0,0 +1,179 @@
+
+
+def parse_inputs(inputs):
+    kwargs = {}
+    if not inputs:
+        return kwargs
+
+    if isinstance(inputs, str):
+        inputs = [inputs]
+
+    for input_ in inputs:
+        key, value = input_.split("=")
+        if "," in value:
+            value = [v for v in value.strip("[]").strip("()").split(",") if v]
+        elif value.lower() in ("true", "false"):
+            value = value == "true"
+        kwargs[key] = value
+    return kwargs
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description=HELP_DESCRIPTION, formatter_class=argparse.RawTextHelpFormatter
+    )
+    parser.add_argument("file_path", type=str, help="Path to the data file")
+    parser.add_argument(
+        "hvplot_kwargs", nargs="*", help="HoloViews options in 'key=value' format"
+    )
+    parser.add_argument(
+        "--opts", nargs="*", help="HoloViews plot options in 'key=value' format"
+    )
+    parser.add_argument(
+        "-r",
+        "--reader",
+        type=str,
+        help="Fully-qualified name of the reader function to use, e.g. pandas.read_csv",
+    )
+    parser.add_argument(
+        "-rk",
+        "--reader_kwargs",
+        type=str,
+        help="Reader options in 'key=value' format",
+    )
+    parser.add_argument(
+        "--port",
+        type=int,
+        default=5006,
+        help="Port to use for displaying the plot",
+    )
+    parser.add_argument(
+        "-o", "--output_path", type=str, help="Path to save the output file"
+    )
+    return parser.parse_args()
+
+
+def main():
+    args = parse_arguments()
+
+    # Process file extension
+    path, extension = os.path.splitext(args.file_path)
+    setup_extension_readers(extension)
+
+    # Determine and use the appropriate file reader
+    if args.reader:
+        module_name, reader_name = args.reader.rsplit(".", 1)
+        import_and_register_module(
+            module_name, extensions=[extension], default_reader=reader_name
+        )
+    try:
+        reader, reader_kwargs = _EXTENSION_READERS[extension]
+    except KeyError:
+        raise NotImplementedError(f"Extension {extension} not supported")
+    reader_kwargs.update(parse_inputs(args.reader_kwargs))
+    data = reader(args.file_path, **reader_kwargs)
+
+    # Process hvplot kwargs
+    hvplot_kwargs = parse_inputs(args.hvplot_kwargs)
+    hvplot_kwargs["title"] = hvplot_kwargs.get("title", path)
+    hvplot_kwargs["kind"] = hvplot_kwargs.get("kind", "explorer")
+    for key in ["groupby", "by", "features"]:
+        value = hvplot_kwargs.get(key, None)
+        if value and not isinstance(value, list):
+            hvplot_kwargs[key] = [value]
+
+    # Process opts
+    opts_kwargs = parse_inputs(args.opts)
+    if opts_kwargs and hvplot_kwargs["kind"] == "explorer":
+        hvplot_kwargs["opts"] = opts_kwargs
+
+    plot = data.hvplot(**hvplot_kwargs)
+    if opts_kwargs and hvplot_kwargs["kind"] != "explorer":
+        plot = plot.opts(**opts_kwargs)
```

Comment:
I can't quite follow the logic over these few lines, but does it achieve making the Explorer just as configurable as a non-explorer plot (with HoloViews options supported, not just hvPlot keywords)? If not, can that be supported (both for Explorer in Jupyter and Explorer via the CLI)?

Reply:
hvplot_explorer is not a native HoloViews object (it's a HvDataFrameExplorer for example) so it doesn't accept an opts method. In the near future, I imagine we can add an opts method to forward the opts rather than these logic.

Reply:
Ok, makes sense!